src/main.rs

```diff
@@ -56,6 +60,10 @@ impl<'a> Mqtt<'a> {
         mqtt_options.set_keep_alive(self.keep_alive);
         mqtt_options.set_clean_session(self.clean_session);
 
+        if self.user.is_some() && self.password.is_some() {
+            mqtt_options.set_credentials(self.user.unwrap(), self.password.unwrap());
```

Comment:
This doesn't seem very idiomatic - calling `is_some()` and then `unwrap()`. I need to think about an alternative, though - as I'm also not super fluent yet on those things. 

If you have an idea, feel free to also chime in.

Reply:
Can you test whether [`unwrap_or_default()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_default) will work?

I think it should. `rumqttc` defaults `""` to `None` for the user, also uses `unwrap_or_default()` for `password: String`, which should result in `""` as a default, too: https://docs.rs/rumqttc/latest/src/rumqttc/lib.rs.html#733

I think that could be clean & easy:

```rust
        mqtt_options.set_credentials(
            self.user.unwrap_or_default(),
            self.password.unwrap_or_default(),
        );
```

Reply:
Also would've the advantage that a setup of a user without a password would work (if that's even a possibility with an mqtt server)

Reply:
If I'm reading the rumqttc library correctly, that function you're referencing is trying to generate a MqttOptions from a URL, and if the username from the URL is the empty string, then sets the MqttOptions credentials field to None.  Digging deeper into the library, if the user/pass is some (including the empty string), a login struct gets created and passed into the creation of the network connection:
https://docs.rs/rumqttc/latest/src/rumqttc/eventloop.rs.html#375

I think I've found a more idiomatic way of doing this though, take a look.

Reply:
Nice, looking good!